src/coreclr/jit/compiler.cpp

```diff
@@ -5266,6 +5268,14 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     fgDebugCheckLinks();
 #endif
 
+#if FEATURE_LOOP_ALIGN
+    if (loopAlignCandidates > 0)
```

Comment:
IMO, this check should happen within the phase, and just early out. Makes the "list of phases" part of the code simpler.